lib/nostrum/api.ex

```diff
@@ -136,158 +138,215 @@ defmodule Nostrum.Api do
   end
 
   @doc ~S"""
-  Send a message to a channel.
+  Posts a message to a guild text or DM channel.
 
-  ## Parameters
-    - `channel_id` - Id of the channel to send the message to.
-    - `content` - One of string, embed, or file to send to the channel.
-    See `t:message_content/0` for more info.
-    - `tts` - Whether the message should be read over text to speech.
+  This endpoint requires the `VIEW_CHANNEL` and `SEND_MESSAGES` permissions. It
+  may situationally need the `SEND_MESSAGES_TTS` permission. It fires the
+  `t:Nostrum.Consumer.message_create/0` event.
+
+  If successful, returns `{:ok, message}`. Otherwise, returns a `t:Nostrum.Api.error/0`.
 
-  For the `channel_id` parameter, you can pass in a `Nostrum.Struct.Message`
-  struct, and it will pull the id from there.
+  ## Options
+
+    * `:content` (string) - the message contents (up to 2000 characters)
+    * `:nonce` (`t:Nostrum.Struct.Snowflake.t/0`) - a nonce that can be used for
+    optimistic message sending
+    * `:tts` (boolean) - true if this is a TTS message
+    * `:file` (`t:Path.t/0`) - the path of the file being sent
+    * `:embed` (`t:Nostrum.Struct.Embed.t/0`) - embedded rich content
+
+    At least one of the following is required: `:content`, `:file`, `:embed`.
+
+  ## Examples
 
-  ## Example
   ```Elixir
-  Nostrum.Api.create_message(1111111111111, [content: "my os rules", file: ~S"C:\i\use\windows"])
+  Nostrum.Api.create_message(43189401384091, content: "hello world!")
+
+  import Nostrum.Struct.Embed
+  embed =
+    %Nostrum.Struct.Embed{}
+    |> put_title("embed")
+    |> put_description("new desc")
+  Nostrum.Api.create_message(43189401384091, embed: embed)
+
+  Nostrum.Api.create_message(43189401384091, file: "/path/to/file.txt")
+
+  Nostrum.Api.create_message(43189401384091, content: "hello world!", embed: embed, file: "/path/to/file.txt")
   ```
   """
-  @spec create_message(Message.t(), message_content, boolean) :: error | {:ok, Message.t()}
```

Comment:
We should remove the `message_content` type at the top of this file as it won't be used anymore.